Sources/MongoSwift/BSON/Document+Sequence.swift

```diff
@@ -17,6 +17,7 @@ extension Document: Sequence {
     // instead of something like `Slice<Document>`.
     /// The type that is returned from methods such as `dropFirst()` and `split()`.
     public typealias SubSequence = Document
+    public typealias Index = Int
```

Comment:
it feels kind of odd that this is defined here, but only used in `Document+Collection`. it looks like it is part of the `Collection` protocol rather than `Sequence` from the [docs](https://developer.apple.com/documentation/swift/collection/2943866-index). should it be defined in that extension instead?

Reply:
Sure!